### ISSUE TITLE
Performances

### ISSUE THREAD

**Issue by Diego999** [danielegrattarola/keras-gat#7]
Hi,

I implemented the GAT model in pytorch where I obtain 83.6-84.6 (https://github.com/Diego999/pyGAT).

I was comparing your code just to double check what I did and have observed that you only do averaging while you should do concat for the n attention heads and average for the last layer (output layer). Maybe your lack of performances is from here (in fact I'm pretty sure because I got similar results before).

**Comment by danielegrattarola** [comment]
If you look at the [model definition](https://github.com/danielegrattarola/keras-gat/blob/b680ec06dadbc8a3c7d9d2cc94cddb18fa164635/examples/gat.py#L37-L49) you can se that indeed I concatenate the heads in the first layer and average them in the second (look at the `attn_heads_reduction` argument of the gat layers).

The lower performance reported in the README is probably due to some variance in the model, unless I misunderstood your point and there is actually a mistake in the code (highly probable).

**Comment by Diego999** [comment]
Hum I thought that using graph_attention_1 in the regularizer is not enough (I never used Keras). In my case, I obtained similar performances with what I mentioned.

**Comment by danielegrattarola** [comment]
I'll look into it, you might be absolutely right. 
[Here](https://github.com/danielegrattarola/keras-gat/blob/b680ec06dadbc8a3c7d9d2cc94cddb18fa164635/keras_gat/graph_attention_layer.py#L125-L131) is the code responsible for the concat/average reduction btw, if you want to check it out.

**Comment by Diego999** [comment]
Yeah I’ve seen you coded the both operations. I’ll check it out later (at least try)

**Comment by AYG-DL** [comment]
Hi @Diego999 @danielegrattarola ,

Is there any update? I tried to run the code and still get similar performance.
